plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -176,15 +184,18 @@ def save(self):
         """
         Save the content and related artifacts in a single DB transaction.
         Due to race conditions, the content may already exist raising an IntegrityError.
-        When this happens, the model is fetched and replaced.
+        When this happens, the model is fetched and _stored_model is updated.
         """
         with transaction.atomic():
-            try:
-                with transaction.atomic():
-                    self.model.save()
-            except IntegrityError:
-                content = type(self.model).objects.get(**self.key)
-                self.update(content)
+            if not self._stored_model:
```

Comment:
Why is this check needed? To decrease reads from DB (`IntegrityError` case)?

Reply:
Pretty much.  Having the `_stored_model` set indicates that the content already existed in the DB.  Saving would certainly cause `IntegretyError` which would result in an unnecessary update.  Both the insert and update would be unnecessary.  The `IntegretyError` is there to catch race conditions whereby the content didn't exist when the changeset looked but exists when another process creates it.